### ISSUE TITLE
Fix the functional test: wallet_keypool.py

### ISSUE THREAD

**Issue by TheQuantumPhysicist** [ghost-coin/ghost-legacy#5]

**Comment by akshaynexus** [comment]
Error for ref
```152/175 - wallet_importmulti.py passed, Duration: 73 s
153/175 - wallet_keypool.py failed, Duration: 21 s

stdout:
2020-05-21T18:38:24.929000Z TestFramework (INFO): Initializing test directory /tmp/particl_test_runner_P_🏃_20200521_211256/wallet_keypool_20
2020-05-21T18:38:43.259000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/akshay/Documents/GitRepos/ghost-private/test/functional/test_framework/util.py", line 122, in try_rpc
    fun(*args, **kwds)
  File "/home/akshay/Documents/GitRepos/ghost-private/test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/home/akshay/Documents/GitRepos/ghost-private/test/functional/test_framework/authproxy.py", line 144, in __call__
    raise JSONRPCException(response['error'], status)
test_framework.authproxy.JSONRPCException: Signing transaction failed (-4)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/akshay/Documents/GitRepos/ghost-private/test/functional/test_framework/test_framework.py", line 114, in main
    self.run_test()
  File "/home/akshay/Documents/GitRepos/ghost-private/test/functional/wallet_keypool.py", line 164, in run_test
    assert_raises_rpc_error(-4, "Transaction needs a change address, but we can't generate it. Please call keypoolrefill first.", w2.walletcreatefundedpsbt, inputs=[], outputs=[{addr.pop(): 0.00005000}], options={"subtractFeeFromOutputs": [0], "feeRate": 0.00010})
  File "/home/akshay/Documents/GitRepos/ghost-private/test/functional/test_framework/util.py", line 114, in assert_raises_rpc_error
    assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
  File "/home/akshay/Documents/GitRepos/ghost-private/test/functional/test_framework/util.py", line 128, in try_rpc
    raise AssertionError(
AssertionError: Expected substring not found in error message:
substring: 'Transaction needs a change address, but we can't generate it. Please call keypoolrefill first.'
error message: 'Signing transaction failed'.
2020-05-21T18:38:43.311000Z TestFramework (INFO): Stopping nodes
2020-05-21T18:38:45.868000Z TestFramework (WARNING): Not cleaning up dir /tmp/particl_test_runner_P_🏃_20200521_211256/wallet_keypool_20
2020-05-21T18:38:45.868000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/particl_test_runner_P_🏃_20200521_211256/wallet_keypool_20/test_framework.log
2020-05-21T18:38:45.869000Z TestFramework (ERROR): Hint: Call /home/akshay/Documents/GitRepos/ghost-private/test/functional/combine_logs.py '/tmp/particl_test_runner_P_🏃_20200521_211256/wallet_keypool_20' to consolidate all logs```

**Comment by TheQuantumPhysicist** [comment]
I've investigated the issue and found that it comes from the variable fSpendable  having an unstable value. At the times of the test, it's true in clang and false in gcc. This has nothing to do with the compiler, as it seems to be some kind of undefined behavior.

This happens in 
```
std::vector<OutputGroup> groups = GroupOutputs(vCoins, !coin_control.m_avoid_partial_spends, max_ancestors);
```
when selecting coins to spend.

Given that this is stateful programming, there's no easy way to track how fSpendable gets an undefined behavior value. I recommend we use clang-sanitizers to fix this issue.

**Comment by akshaynexus** [comment]
This was fixed by #9 being merged
